packages/gatsby/src/schema/graphql-engine/print-plugins.ts

```diff
@@ -0,0 +1,214 @@
+/* eslint @typescript-eslint/no-unused-vars: ["error", { "ignoreRestSiblings": true }] */
+import * as fs from "fs-extra"
+import * as path from "path"
+import * as _ from "lodash"
+import { slash } from "gatsby-core-utils"
+import { store } from "../../redux"
+import { IGatsbyState } from "../../redux/types"
+import { requireGatsbyPlugin } from "../../utils/require-gatsby-plugin"
+
+const schemaCustomizationAPIs = new Set([
+  `setFieldsOnGraphQLNodeType`,
+  `createSchemaCustomization`,
+  `createResolvers`,
+])
+
+const excludePlugins = new Set([`internal-data-bridge`, `default-site-plugin`])
+const includePlugins = new Set([`gatsby-plugin-sharp`])
+
+// Emit file that imports required node APIs
+const schemaCustomizationPluginsPath =
+  process.cwd() + `/.cache/query-engine-plugins.js`
```

Comment:
we usually don't use process.cwd() perhaps we need a global getter for program.directory?

Reply:
We do it all over the place in the codebase. I think it makes sense to add such utility and refactor in a separate PR.